### PR TITLE
Add test case for forkless FLUSHDB/FLUSHALL duplicate reply

### PR DESCRIPTION
**Pull request by enjoy-binbin**
Add test case for #4610. #4610 replaces this fix, but does not
pick the test case. Adding a test case would be harmless and can
provide extra code coverage.

### CODE REVIEW

File: src/bgiteration.c [hunk 1]
```diff
@@ -2315,6 +2320,7 @@ bool bgIteration_blockClientIfRequired(client *c) {
     iteratorReplicationFlagsWereUpdated = false;
     if (!bgIteration_iterationActive()) return false;
     if (!isWriteCmd(c->cmd)) return false;
+    if (isFlushCmd(c->cmd) && (getFlushCommandFlags(c, false, NULL) == C_ERR)) return false;
```
Comment (enjoy-binbin) [inline src/bgiteration.c]: Didn't bother to change the name, using a name like parse* should be better here.

Comment (JimB123) [reply src/bgiteration.c]: Hi @enjoy-binbin - thanks for catching this.  I'd like to tweak the solution a bit and avoid impacting the non-bgiteration code.  I'll open a new PR and send it back to you for review.

Comment (JimB123) [reply src/bgiteration.c]: you're right, it's more consistent to use parse*OrReply here.  I'll fix that too.
